component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -530,8 +530,13 @@ private static boolean isJWKSEnabled() {
     private static boolean validateUsingJWKSUri(SignedJWT signedJWT, String jwksUri) throws IdentityOAuth2Exception {
 
         JWKSBasedJWTValidator jwksBasedJWTValidator = new JWKSBasedJWTValidator();
-        return jwksBasedJWTValidator.validateSignature(signedJWT.getParsedString(), jwksUri,
-                signedJWT.getHeader().getAlgorithm().getName(), null);
+        try {
+            return jwksBasedJWTValidator.validateSignature(signedJWT.getParsedString(), jwksUri,
+                    signedJWT.getHeader().getAlgorithm().getName(), null);
+        } catch (IdentityOAuth2Exception e) {
+            log.error("Error occurred while validating the signature using JWKS URI.", e);
+            throw e;
```

Comment:
Ideally, the error should be logged at the last place of handling. since we are throwing the exception again from here, whats the reason to have the log available here?